ledger/participant-integration-api/src/main/scala/platform/apiserver/execution/StoreBackedCommandExecutor.scala

```diff
@@ -49,10 +49,16 @@ private[apiserver] final class StoreBackedCommandExecutor(
       ec: ExecutionContext,
       loggingContext: LoggingContext,
   ): Future[Either[ErrorCause, CommandExecutionResult]] = {
+    val interpretationTimeNanos = new AtomicLong(0L)
```

Comment:
Why do we need atomic when these computations are serial?

Reply:
If I understand correctly, it is used only to pass a reference to a mutable value to trackSyncExecution. We could rewrite it, but I'm not sure which one is better:

```scala
var interpretationTimeNanos: Long = 0L

trackSyncExecution(dt => interpretationTimeNanos += dt)(...)

private def trackSyncExecution[T](updateTime: Long => ())(computation: => T): T = {
    val start = System.nanoTime()
    val result = computation
    updateTime(System.nanoTime() - start)
    result
}
```

I would at least add a comment explaining that interpretation steps are serial and why AtomicLong was chosen.

Reply:
we do not need it strictly speaking, just all the rest of these cumulated queries expressed that way. For this approach we need protection, since the metrics population happens asynchronously.
What we could do is to switch to async-recursion accumulators instead, and populate the metrics in the final pattern matching clauses, it has its advantages, but also more intrusive to main intent (the engine looping).
For me it does not make much sense to have two approaches here, happy to refactor if you think it is much better, I have no strong opinion

Reply:
:+1: